.github/workflows/nodejs.yml

```diff
@@ -70,6 +70,93 @@ jobs:
       runs-on: ${{ matrix.runs-on }}
     secrets: inherit
 
+  test-external:
+    name: Test with Node.js ${{ matrix.version }} compiled --shared-builtin-undici/undici-path
+    strategy:
+      fail-fast: false
+      max-parallel: 0
+      matrix:
+        version: [20, 22]
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    steps:
+      # Checkout into a subdirectory otherwise Node.js tests will break due to finding Undici's package.json in a parent directory.
```

Comment:
FWIW Node.js tests that break if a `package.json` is found in a parent directory of Node.js' tests:
- Node.js 20: `test-policy-*`. The policy feature is removed from Node.js 22 and main so the tests don't exist there.
- Node.js 22: `test-node-run`.